mitodl/ocw-hugo-themes
# feat: keyboard navigable video buttons

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #1126 

#### What's this PR do?
This PR: 
- Makes video buttons (such as transcript, related resources, download icons, etc) keyboard navigable
- Introduces playwright tests for the keyboard-navigable functionality.

#### How should this be manually tested?
**Testing functionality:**
During these steps, make sure the mobile responsiveness for the video page/video player also works as intended.
1. Checkout to this branch
2. `yarn start course res.18-009-fall-2015`
3. For embed videos:
    - Go to this link: http://localhost:3000/pages/video-series-overview/
    - Navigate by using `'tab'` key and go to video page by using tabs and then enter at `"VIEW VIDEO PAGE"` tab bar. 
4. Stop the local server and use this course now: `yarn start course res.18-009-fall-2015`
5. Go to any video page: http://localhost:3000/resources/part-1-introduction-and-overview-of-business-plans/
6. Using `tabs` in combination of `enter` or `spacebar`, make sure you can:
    - Collapse/expand the tab bars (for Transcript / Lecture Notes)
    - For Lecture Notes, make sure you can navigate using keyboard and open the link inside text: `Introduction to Business Plans (PDF)`
    - Make sure you can navigate to the download video icon and the links at "Download video" "Download transcript" are keyboard navigable and clickable.
 
**Running playwright tests:**

1. Checkout to this branch.
2. Run each playwright test about 200 times to ensure there is no flakiness:
`yarn test:e2e video-tabs.spec.ts --repeat-each 200
`

### Reviewing Playwright tests:
For reviewing playwright tests of this PR, this discussion might be helpful as it also relates to this PR:
https://github.com/mitodl/hq/discussions/1391
